plugins/rest/rest.go

```diff
@@ -354,15 +359,14 @@ func (c Client) Do(ctx context.Context, method, path string) (*http.Response, er
 	return resp, err
 }
 
-func withMaskedAuthorizationHeader(headers http.Header) http.Header {
-	authzHeader := headers.Get("Authorization")
-	if authzHeader != "" {
-		masked := make(http.Header)
-		for k, v := range headers {
+func withMaskedAuthorizationHeaders(headers http.Header) http.Header {
+	masked := make(http.Header)
+	for k, v := range headers {
+		if _, ok := maskedHeaderKeys[k]; ok {
+			masked.Add(k, "REDACTED")
```

Comment:
Nit: We could do a `masked.Set` which replaces any existing values. Maybe not a concern here.

Reply:
I'll make this change, thanks for calling it out.  I'm not sure it's a concern, since it's adding to a fresh instance of `http.Header`, but I'm always in favor of being more explicit 👍